AUTH48 Reply

wtoorop opened the pull request.

wtoorop committed:
AUTH48 Reply

peterthomassen reviewed (commented):
Added some comments and think we can accept all other suggested changes collectively.

The IANA remarks at the end need specific answers, but I'm not an expert at that.

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+> Original:
+>    C.R. Monshouwer -->
+> 
+> 
```
```suggestion
The original is correct. The initials should be C.R.
> 
```

@mind04 Please correct me if I'm wrong

oerdnj commented:
I’ve already read the diff from IANA and was ok with changes, but seems like you have keener eyes.

Habbie commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+> Perhaps: 
+>    This document makes use of terminology for transfer mechanisms (AXFR
+>    and IXFR), record types (SOA, NS, and PTR), and other technical terms (such as
+>    RDATA) that are specific to the DNS. -->
```
works for me

Habbie commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+>    The names of member zones are represented on the RDATA side of a PTR
+>    record (instead of being represented as a part of owner names) so that all
+>    valid domain names may be represented regardless of their length
+>    [RFC1035]. -->
```
works for me

Habbie commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+> 
+> Perhaps:
+>    Note that Version 1 was used in an earlier draft version of this memo 
+>    and reflected the implementation first found in BIND 9.11. -->
```
```suggestion
>    and reflected the implementation first found in BIND 9.11. -->

Please replace with your proposal, thank you.
```

Habbie commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+> 
+> 15) <!-- [rfced] Should we change "QUERY" to "query" in this context? Or should
+> "QUERY" remain capitalized due to its association with "DNS QUERY" in the
+> previous sentence?
```
I vote for Original

wtoorop replied:
I have no opinion on this, so ok, the original

wtoorop committed:
Merge branch 'master' into AUTH48-reply

wtoorop committed:
Set all sourcecode types to dns-rr

wtoorop committed:
Results of reviewing <tt> and <em> formatting

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+>    with each operator ("operator-x-foo" vs. "operator-y" "bar"). -->
+> 
+> 
+> 12) <!-- [rfced] FYI - We have rearranged this sentence for readability. Please
+> let us know if this change is incorrect or if there are any objections.
+> 
+> Original: 
+>    Only when the zone was configured from a specific catalog zone, and
+>    the zone is removed as a member from that specific catalog zone, the zone and
+>    associated state (such as zone data and DNSSEC keys) MUST be removed from the
+>    consumer.
+> 
+> Current: 
+>    The zone and associated state (such as zone data and DNSSEC keys)
+>    MUST be removed from the consumer only when the zone was configured from a
+>    specific catalog zone and is removed as a member from that specific catalog
+>    zone. -->
```
Actually the MUST NOT case was in the sentence preceding this one.  We either maintain the original text, or change into something like this (which is I think clearer): 

The zone and associated state (such as zone data and DNSSEC keys) MUST be removed from the consumer only when the zone was configured initially from the same catalog.

@peterthomassen @Habbie WDYT?

libor-peltan-cznic replied:
How about using the latter with "when and only when" ?

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+>    including the removal of all the zone's associated state (as described in 
+>    Section 5.3), and then immediately process the member as a newly added zone 
+>    to be configured in the same catalog.-->
+> 
```
```suggestion
> 

Works for me!
```

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+>    In the process of a migration, if some consumers of the involved
+>    catalog zones did not catch the removal of the member zone from
+>    $OLDCATZ yet (because of a lost packet, downtime, or otherwise) but
+>    already saw the update of $NEWCATZ, they may consider the update by
```
indeed, maybe we can rearrange "the update adding the member zone" into "the adding-the-member-zone update" WDYT @peterthomassen @Habbie

libor-peltan-cznic replied:
Yes. I would perhaps simplify this difficult sentence by removing the redundant "adding the member zone", like this: "...they may consider this update to be a name clash..."

peterthomassen replied:
The update also appears earlier in the sentence, and when qualify the word "update" there, things get easier:

```
    If in the process of a migration some consumers of the involved
    catalog zones did not catch the removal of the member zone from
    $OLDCATZ yet (because of a lost packet or downtime or otherwise), but
    did already see the update of $NEWCATZ containing the addition of that
    member zone, they may consider this update to be a name clash (see
    Section 5.2) and as a consequence the member is not migrated to
    $NEWCATZ. 
```

libor-peltan-cznic commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+>    and reflected the implementation first found in BIND 9.11. -->
+> 
+> 
+> 9) <!-- [rfced] We suggest rephrasing the following sentence to provide a better
+> flow of the text and avoid interrupters. Does the following suggestion
+> retain the intended meaning?
+> 
+> Original: 
+>    The consumer MUST verify, before the actual migration, that coo
+>    property pointing to $NEWCATZ is still present in $OLDCATZ.
+> 
+> Perhaps: 
+>    The consumer MUST verify that the coo property pointing to $NEWCATZ
+>    is still present in $OLDCATZ before the actual migration. -->
```
Agreed. How about moving "before the actual migration" to the beginning of the sentence?

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+> This document (current):
+>    This registry does not apply to catalog zones version "1" but applies 
+>    to catalog zones version "2" as specified in RFC 9432.
+> 
```
```suggestion
> 
I vote for what is currently in the registry.
```

@Habbie @peterthomassen @libor-peltan-cznic @mind04 WDYT?

peterthomassen replied:
ack

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+>    Status: IETF Stream RFC status or "External" if not documented in an
+>    IETF Stream RFC.
+> -->
+> 
```
```suggestion
> 
Yes, that's fine with me.
```

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+> used inconsistently. To match RFCs 6761, 8945, and 9103, we have updated
+> each instance of "nameserver" to "name server". Please let us know any
+> objections. -->
+> 
```
```suggestion
> 
This is fine! Thanks
```

peterthomassen commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+> updated for clarity.
+> 
+> Original: 
+>    The group value that indicates that depends on what has been agreed
+>    with each operator ("operator-x-foo" vs. "operator-y" "bar"). -->
```
sounds good, except that the comma does not belong there

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,442 @@
+>    the contents of a multivalued property (such as the list of member
+>    zones) with a single query. -->
+> 
+> 
```
```suggestion
> 

We prefer the original (and kept it in the attached xml).
```

wtoorop committed:
Apply suggestions from code review

All the cases in which we either accept the change made by the editor, or stick with the old text which is still in the xml.
I will commit the xml after this commit.

wtoorop committed:
Remove <-- rfced --> in xml where we chose the text to remain

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,463 @@
+>    This document makes use of terminology for transfer mechanisms (AXFR
+>    and IXFR), record types (SOA, NS, and PTR), and other technical terms (such as
+>    RDATA) that are specific to the DNS. -->
+> 
```
```suggestion
> 

Thanks, we accept the new text and have updated the attached xml with the suggestion.
```

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,463 @@
+>    record (instead of being represented as a part of owner names) so that all
+>    valid domain names may be represented regardless of their length
+>    [RFC1035]. -->
+> 
```
```suggestion
> 

Thanks, we accept the new text and have updated the attached xml with the suggestion.
```

wtoorop committed:
Apply suggestions from code review

A batch in which we take the suggestions.

Co-authored-by: Peter van Dijk <peter.van.dijk@powerdns.com>

wtoorop committed:
The cases where we take the suggestion from the rfc-editor

wtoorop committed:
Merge branch 'AUTH48-reply' of github.com:NLnetLabs/draft-toorop-dnsop-dns-catalog-zones into AUTH48-reply

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,476 @@
+> Perhaps: 
+>    The consumer MUST verify that the coo property pointing to $NEWCATZ
+>    is still present in $OLDCATZ before the actual migration. -->
+> 
```
```suggestion
> 

We don't take the suggestion but have updated the xml with: "Before the actual migration the consumer MUST verify that the coo property pointing to $NEWCATZ is still present in $OLDCATZ."

Is that also more clear in your opinion?

```

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,476 @@
+>    MUST be removed from the consumer only when the zone was configured from a
+>    specific catalog zone and is removed as a member from that specific catalog
+>    zone. -->
+> 
```
```suggestion
> 

FYI - We modified your rearrangement a bit in the attached updated xml, to say: The zone and associated state (such as zone data and DNSSEC keys) MUST be removed from the consumer when and only when the zone was configured initially from the same catalog.

```

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,476 @@
+>    adding the member zone in $NEWCATZ to be a name clash (see
+>    Section 5.2); as a consequence, the member is not migrated to
+>    $NEWCATZ. -->
+> 
```
```suggestion
> 

FYI - we updated your readability update to the following in the attached xml: If in the process of a migration some consumers of the involved catalog zones did not catch the removal of the member zone from $OLDCATZ yet (because of a lost packet or downtime or otherwise) but already saw the update of $NEWCATZ containing the addition of that member zone, they may consider this update to be a name clash (see Section 5.2) and as a consequence the member is not migrated to $NEWCATZ. 

```

wtoorop commented on AUTH48-reply.txt:
```diff
@@ -0,0 +1,476 @@
+> us know if any changes are needed. Note that our script did not flag any
+> words in particular, but this should still be reviewed as a best
+> practice. -->
+> 
```
```suggestion
> 

After going over the text once again, I didn't see any issues with respect to "Inclusive Language"

```

wtoorop committed:
Apply suggestions from code review

All the things where we changed the suggestions or text updated by the rfc-editor

wtoorop committed:
All the cases in which we changed the rfc-editor suggestions or updated texts

wtoorop committed:
Fix typo found with getting it through xml2rfc

mind04 reviewed (changes requested):
One nit, otherwise the document has my approval

wtoorop committed:
AUTH48 review feedback from @mind04 s/associated states/associated state/g

wtoorop committed:
Notify rfc editor about changes in commit @3955427

wtoorop commented on AUTH48-rfc9432.xml:
```diff
@@ -304,35 +229,10 @@ The presence of more than one record in the RRset indicates a broken catalog zon
-   is still present in $OLDCATZ before the actual migration. -->
-The consumer <bcp14>MUST</bcp14> verify, before the actual migration, that <tt>coo</tt> property pointing to <tt>$NEWCATZ</tt> is still present in <tt>$OLDCATZ</tt>.</t>
+Before the actual migration the consumer <bcp14>MUST</bcp14> verify that the <tt>coo</tt> property pointing to <tt>$NEWCATZ</tt> is still present in <tt>$OLDCATZ</tt>.</t>
 <t>Unless the member node label (i.e., <tt>&lt;unique-N&gt;</tt>) for the member is the same in <tt>$NEWCATZ</tt>, all its associated states for a just migrated zone <bcp14>MUST</bcp14> be reset (see <xref target="zonereset"></xref>).
```
Thanks! Addressed in commit 39554276
